fix LayoutInitializer to handle null pipeline and add MacroRunner particle+nodetype

mariakleiner opened the pull request:
and avoid outputting an unchanged pipeline in ConnectionUpdator.

sjmiles commented on pkg/Library/NodeGraph/ConnectionUpdator.js:
```diff
@@ -17,15 +17,17 @@ update(inputs, state) {
     const {pipeline, nodeTypes, candidates} = inputs;
     assign(state, {pipeline, candidates});
     if (candidates) {
-      //let changed = false;
```
maybe 

let changed = this.removePipeline...
changed ||= this.updatePiepline...

mariakleiner replied:
`change || = doSomething()` will not call `doSomething`, if `changed` is already `true`.

sjmiles replied:
Touché

mariakleiner commented:
sorry, i completely messed up LayoutInitializer again (didn't work for new pipelines), so I fixed it (hopefully) and also added the MacroRunner to the same PR.
